text/0085-api-keys.md

```diff
@@ -0,0 +1,463 @@
+
+##### List details
+
+- `API Key` objects are returned in a `results` array.
+- `API Keys` are ordered by `createdAt` in `desc` order. (Most recent first)
+- Expired `API Keys` are omitted and no longer accessible. See Future Possibilities part.
+- No pagination yet. See Future Possibilities part.
+
+##### Errors
+
+- 🔴 Accessing this route without the `X-MEILI-API-KEY` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 Accessing this route with a key that does not have permissions (i.e. other than the main-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+---
+
+#### **As `Anna 👩`, I want to delete an `API Key`**
+
+##### Request Definition
+
+`DELETE` - `/keys/:key`
+
+##### Headers
+
+```
+"X-MEILI-API-KEY: :mainKey"
+```
+
+##### Response
+
+`204 No-Content`
+
+##### Errors
+
+- 🔴 Accessing this route without the `X-MEILI-API-KEY` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 Accessing this route with a key that does not have permissions (i.e. other than the main-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+- 🔴 Attempting to access an API key that does not exist returns a `api_key_not_found`.
+
+---
+
+#### Using `API Key` on client-code
+
+![](https://i.imgur.com/kRKcB43.png)
+
+`Mark 👨🏻` receives the `API Key` transmitted by `Anna 👩` on a secured channel of their choice. He uses it to authenticate requests from the client code to MeiliSearch.
+
+#### **As `Mark 👨🏻`, I am using an expired/deleted API Key**
+
+#### Request example
+
+`POST` - `/indexes/movies/search`
+
+#### Headers
+
+```
+    "X-MEILI-API-KEY": ":apiKey"
+    "Content-Type: application/json"
+```
+
+#### Response
+
+- 🔴 Accessing this route with an `API Key` that has expired or, been deleted returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+---
+
+### **As `Mark 👨🏻`, I am using a valid API Key, but the permission set is not sufficient to access the requested API resource**
+
+#### Request example
+
+`POST` - `/indexes/movies/search`
+
+#### Headers
+
+```
+    "X-MEILI-API-KEY": ":apiKey"
+    "Content-Type: application/json"
+```
+
+#### Response
+
+- 🔴 Accessing this route with an `API Key` that don't have sufficient permissions to access it returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+## 2. Technical Aspects
+
+### Synchronous write of `API Key` resources
+
+Synchronous writing would make it possible to dissociate key management from the update store, which handles asynchronous writing.
```

Comment:
I'm a bit confused: will the API Key endpoint be sync or async? The phrasing makes me interpret this as a suggestion, but I'm assuming I'm wrong?

Reply:
Indeed, it is not clear. Thanks, @guimachiavelli. I rephrased that part to make it clear that the operations on /keys are synchronous.